src/python/WMCore/WorkQueue/DataLocationMapper.py

```diff
@@ -126,6 +129,23 @@ def locationsFromDBS(self, dbs, dataItems):
 
         return result
 
+    def locationsFromPileup(self, dataItems, dbsUrl):
+        """Get data location from MSPileup"""
+        result = defaultdict(set)
+        for dataItem in dataItems:
+            try:
+                queryDict = {'pileupName': dataItem,
+                             'filters': ['currentRSEs', 'pileupName', 'containerFraction', 'ruleIds']}
+                testBed = '-testbed' if 'testbed' in dbsUrl else ''
```

Comment:
And this line I would suggest to replace by something like:
```
                instance = '-testbed' if 'cmsweb-testbed' in dbsUrl else '-prod'
```
It's been a couple of years that we adopted the cmsweb-prod cluster for production activity.